### ISSUE TITLE
Poor word LM results on PTB

### ISSUE THREAD

**Issue by VHellendoorn** [Microsoft/CNTK#2799]
**Caveat:** I am using CUDA 9/CUDNN 7, so the below issue may be due to that configuration.

The [WordLMWithSampledSoftmax](https://github.com/Microsoft/CNTK/tree/master/Examples/Text/WordLMWithSampledSoftmax) Example allows one to run a common "Small" RNN configuration (2 layer 200-dim LSTM) on the PTB dataset. This is a pretty standard setup, in which test/validation entropies should level out around **115 PPL** (_e.g._ in [Tensorflow](https://github.com/tensorflow/models/blob/master/tutorials/rnn/ptb/ptb_word_lm.py)). However, even with full softmax, this implementation doesn't come close to that, instead leveling out around **300 PPL**. This remains true across a range of different configurations, such as without momentum, with simple SGD, lower/higher learning rates, different batch sizes _etc._.

There does not seem to be any other general language modeling tutorial/example or other issue addressing this, nor are the expected outputs published. Are these results expected? What configuration would replicate other "Small" configs for the PTB dataset? Thanks

**Comment by VHellendoorn** [comment]
Any comments on this? Is it a mistake on my part? I've preferred CNTK over Tensorflow so far, but this is preventing me from using it for language modeling tasks.
